### PR TITLE
Access ExitCode from maraboupy 

### PR DESCRIPTION
**Pull request by anwu1219**
1. Now the solve method of Maraboupy returns the exitcode of the Engine. Before there was no way to differ "error" and "unsat".
2. Extended the tests/examples accordingly.
3. decrease the default value of the numerical tolerance.

### CODE REVIEW

File: src/configuration/GlobalConfiguration.cpp [hunk 1]
```diff
@@ -70,8 +70,8 @@ const unsigned GlobalConfiguration::PSE_ITERATIONS_BEFORE_RESET = 1000;
 const double GlobalConfiguration::PSE_GAMMA_ERROR_THRESHOLD = 0.001;
 const double GlobalConfiguration::PSE_GAMMA_UPDATE_TOLERANCE = 0.000000001;
 
-const double GlobalConfiguration::RELU_CONSTRAINT_COMPARISON_TOLERANCE = 0.00001;
-const double GlobalConfiguration::ABS_CONSTRAINT_COMPARISON_TOLERANCE = 0.00001;
+const double GlobalConfiguration::RELU_CONSTRAINT_COMPARISON_TOLERANCE = 1e-6;
```
Comment (AleksandarZeljic) [inline src/configuration/GlobalConfiguration.cpp:73]: This is a slightly smaller value. It might make sense to make this change in a separate PR so that is obvious, as it might affect performance (I think it's unlikely), but it's a cleaner approach in case it does affect performance.

**Review by AleksandarZeljic — APPROVED** [review]
LGTM. See the comment re: GlobalConfiguration values.

**Comment by anwu1219** [comment]
> LGTM. See the comment re: GlobalConfiguration values.

I agree it's cleaner to do this in a different PR in case there's big performance change. Thanks for pointing this out! I reverted that particular change and will submit a separate PR.
